components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SPInitSSOAuthnRequestValidator.java

```diff
@@ -135,18 +130,19 @@ public SAMLSSOReqValidationResponseDTO validate() throws IdentityException {
             }
 
             // subject confirmation should not exist
-            if (subject != null && subject.getSubjectConfirmations() != null && subject.getSubjectConfirmations().size() > 0) {
-                validationResponse.setValid(false);
-                String errorResp =
-                        buildErrorResponse(SAMLSSOConstants.StatusCodes.REQUESTOR_ERROR,
-                                "Subject Confirmation methods should NOT be in the request.");
-                if (log.isDebugEnabled()) {
-                    log.debug("Invalid Request message. A Subject confirmation method found " +
-                            subject.getSubjectConfirmations().get(0));
-                }
-                validationResponse.setResponse(errorResp);
-                validationResponse.setValid(false);
-                return validationResponse;
+            if(subject != null && subject.getSubjectConfirmations() != null &&
+                    subject.getSubjectConfirmations().size() > 0) {
```

Comment:
We should be able to use List.isEmpty instead of comparing the size of the array against 0.